.github/workflows/packages.yml

```diff
@@ -158,6 +30,37 @@ jobs:
       - name: Checkout repos
         uses: actions/checkout@v4
 
+      # The deployment target should be <= to that of the oldest supported Python version.
+      # e.g. https://www.python.org/downloads/release/python-380/
+      # The MACOSX_DEPLOYMENT_TARGET variable must not be exposed to cibuildwheel
+      - run: echo deployment_target=${{ matrix.arch == 'arm64' && '11.0' || '10.9' }} | tee $GITHUB_ENV
+      - run: echo fake_prefix="$PWD/fake-prefix" >> $GITHUB_ENV
+
+      - name: Build openssl from source
+        run: |
+          export MACOSX_DEPLOYMENT_TARGET=${{ env.deployment_target }}
+          curl --fail-with-body -L https://github.com/openssl/openssl/releases/download/openssl-3.4.0/openssl-3.4.0.tar.gz | tar xz
+          cd openssl-3.4.0
+          ./configure --prefix="$fake_prefix"
+          make build_sw
+          make install_sw
+
+      - name: Build postgresql from source
+        run: |
+          curl --fail-with-body -L https://ftp.postgresql.org/pub/source/v16.6/postgresql-16.6.tar.bz2 | tar x
+          brew uninstall openssl --ignore-dependencies
+
+          export MACOSX_DEPLOYMENT_TARGET=${{ env.deployment_target }}
+          export LDFLAGS="-L$fake_prefix/lib"
+          export CPPFLAGS="-I$fake_prefix/include"
+          cd postgresql-16.6
+          ./configure --without-icu --prefix="$fake_prefix" --with-ssl=openssl
```

Comment:
No gssapi/ldap support?

Reply:
Ah, I was going to ask about which `--with-xxx` features to use but you beat me to it. I see `build_libpq.sh` makes that decision for me anyway.

Reply:
Unfortunately that also implies building these other libraries. Not doing so would break certain features. I don't know how many people rely on them, but I don't feel they are features that we can drop without being unpunished.

TBH, for some reason, GSSAPI causes a lot of problems to macOS users because in some configuration it causes slowdowns on connection, but still it doesn't seem a good case to decide that nobody in the world requires it.